cclib/parser/gaussianparser.py

```diff
@@ -213,6 +214,230 @@ def extract(self, inputfile, line):
+        # <L.S>= 0.000000000000E+00
+        # Annihilation of the first spin contaminant:
+        # S**2 before annihilation     1.0237,   after     0.2796
+        elif 'S**2 before annihilation' in line:
+            if not hasattr(self, 'spinexpect'):
+                self.set_attribute('spinexpect', [])
+                self.set_attribute('spinexpectanni', [])
+            self.extend_attribute('spinexpectanni', float(line.strip().split()[-1]))
+            self.extend_attribute('spinexpect', float(line.strip().split()[-3][:-1]))
+
+        # Extract symmetry number, rotational constants and rotational temperatures
+        elif 'Rotational symmetry number' in line:
+            symmno = int(line.strip().split()[3].split(".")[0])
+            self.set_attribute('rotsymno', symmno)
+
+        # Extract Rotational Constants
+        # Example:
+        # Rotational constants (GHZ):           3.13081     1.24272     0.88960
+        # OR for linear molecules:
+        # Rotational constants (GHZ): ************ 12.73690 12.73690
+        # Note: rotational constant will be converted to wavenumber units (1/cm) to standardize across parsers
+        elif line.find('Rotational constants (GHZ):') > -1:
+            try:
+                roconst = [float(line.strip().replace(':', ' ').split()[3]),
+                                float(line.strip().replace(':', ' ').split()[4]),
+                                float(line.strip().replace(':', ' ').split()[5])]
+            except ValueError:
+                if line.find('********') > -1:
+                    roconst = [float(line.strip().replace(':', ' ').split()[4]),
+                                    float(line.strip().replace(':', ' ').split()[5])]
+            self.append_attribute("rotconsts", roconst)
+
+        elif line.find('Rotational temperature ') > -1:
+            rotemp = [float(line.strip().split()[3])]
+            self.append_attribute("rottemps", rotemp)
+        elif line.find('Rotational temperatures') > -1:
+            try:
+                rotemp = [float(line.strip().split()[3]), float(line.strip().split()[4]),
+                            float(line.strip().split()[5])]
+            except ValueError:
+                if line.find('********') > -1:
+                    rotemp = [float(line.strip().split()[4]), float(line.strip().split()[5])]
+            self.append_attribute("rottemps", rotemp)
+
+        # Detects how many times the calculation found a stationary point. This is important
+        # to detect a problem where Gaussian converges during the optimization step but the
+        # following frequency calculation does not converge (https://gaussian.com/faq3/)
+        #     -- Stationary point found.
+        if line.find('Stationary point found') > -1:
+            if not hasattr(self, 'times_converged'):
+                self.set_attribute('times_converged', 1)
+            else:
+                self.set_attribute('times_converged', 2)
```

Comment:
This is a good first implementation, but is not quite robust enough I think. If the job is only a frequency calc with no prior optimisation, `times_converged` will be 1 rather than the expected 2.

```suggestion
        if line.find('Stationary point found') > -1:
            if hasattr(self, "freqdone"):
                self.freqdone = True
```

Reply:
please check the comment above